mine.js

```diff
@@ -11,10 +11,8 @@ const tmp = require('tmp')
 const path = require('path')
 const fs = require('fs')
 const spawn = require('child_process').spawn
-const Web3 = require('web3')
 
-async function checkForModels (mineAddress, contractAddress, ethereumUrl, ipfsUrl) {
-  const web3 = new Web3(new Web3.providers.HttpProvider(ethereumUrl))
+async function checkForModels (mineAddress, contractAddress, web3, ipfsUrl) {
```

Comment:
Not 💯 sure about this change on the interface. I'd prefer an aligned approach
1. pass web3 and ipfs both as initialized connections
2. pass both as URLs (initialize web3 twice)

How do you feel about it @kevinahuber?